install.sh

```diff
@@ -110,13 +110,13 @@ install_nvm_from_git() {
         exit 2
       }
     else
-      command git clone "$(nvm_source)" -b "$(nvm_latest_version)" --depth=1 "${INSTALL_DIR}" || {
+      command git -c advice.detachedHead=false clone "$(nvm_source)" -b "$(nvm_latest_version)" --depth=1 "${INSTALL_DIR}" || {
```

Comment:
what happens in git version v1.7?

Reply:
Both the current git command and the one in this PR fail for other reasons. `-c advice.detachedHead=false` cause no additional failures.

The failure is
```
warning: Remote branch v4.6.0 not found in upstream origin, using HEAD instead
```

In some version between current and 1.7 the ability to use a commitish instead of only a branch for the `-b` option was added, with the man page text changing from

```
       --branch <name>, -b <name>
           Instead of pointing the newly created HEAD to the branch pointed to by the cloned repository’s HEAD, point to <name> branch instead.
           In a non-bare repository, this is the branch that will be checked out.
```
to
```
       --branch <name>, -b <name>
           Instead of pointing the newly created HEAD to the branch pointed to by the cloned repository’s HEAD, point to <name> branch instead.
           In a non-bare repository, this is the branch that will be checked out.  --branch can also take tags and detaches the HEAD at that
           commit in the resulting repository.
```

This was tested with git version 1.7.9.5, provided by 1:1.7.9.5-1ubuntu0.2

Reply:
Hmm, interesting - that suggests that we need to ratchet git support up. Do you know when that ability was added?

Reply:
Sometime before 1.9, which is what Ubuntu 14.04 has. I'm not sure it's a serious issue, a setup that old is likely to have other backwards compatibility problems. If it is a concern, I guess another issue could be opened. I don't want to tackle that here, because this is just a simple backwards compatible change.

Reply:
Looks like v1.17.10 - https://github.com/git/git/commit/5a7d5b683f869d3e3884a89775241afa515da9e7 - so pretty close.

I'll update your branch so the readme reflects that, and then this can go in. Thanks!